dotnetv3/Cognito/Actions/CognitoWrapper.cs

```diff
@@ -0,0 +1,362 @@
+
+        do
+        {
+            response = await _cognitoService.ListUsersAsync(request);
+            users.AddRange(response.Users);
+            request.PaginationToken = response.PaginationToken;
+        }
+        while (response.PaginationToken is not null);
+
+        return users;
+    }
+
+    // snippet-end:[Cognito.dotnetv3.ListUsers]
+
+    // snippet-start:[Cognito.dotnetv3.AdminRespondToAuthChallenge]
+    public async Task<AuthenticationResultType> AdminRespondToAuthChallengeAsync(string userPoolId, string userName, string clientId, string mfaCode, string session)
+    {
+        var challengeResponses = new Dictionary<string, string>();
+        challengeResponses.Add("USERNAME", userName);
+        challengeResponses.Add("SOFTWARE_TOKEN_MFA_CODE", mfaCode);
+
+        var request = new AdminRespondToAuthChallengeRequest
+        {
+            ClientId = clientId,
+            UserPoolId = userPoolId,
+            ChallengeResponses = challengeResponses,
+            Session = session
+        };
+
+        var response = await _cognitoService.AdminRespondToAuthChallengeAsync(request);
+        return response.AuthenticationResult;
+    }
+
+    // snippet-end:[Cognito.dotnetv3.AdminRespondToAuthChallenge]
+
+    // snippet-start:[Cognito.dotnetv3.RespondToAuthChallenge]
+    /// <summary>
+    /// Respond to an authentication challenge.
+    /// </summary>
+    /// <param name="userName">The name of the user.</param>
+    /// <param name="clientId">The client Id</param>
+    /// <param name="mfaCode">The multi-factor authentication code.</param>
+    /// <param name="session">The current application session.</param>
+    /// <returns>An async Task</returns>
+    public async Task<AuthenticationResultType> RespondToAuthChallengeAsync(string userName, string clientId, string mfaCode, string session)
+    {
+        Console.WriteLine("SOFTWARE_TOKEN_MFA challenge is generated");
+
+        var challengeResponses = new Dictionary<string, string>();
+        challengeResponses.Add("USERNAME", userName);
+        challengeResponses.Add("SOFTWARE_TOKEN_MFA_CODE", mfaCode);
+
+        var respondToAuthChallengeRequest = new RespondToAuthChallengeRequest
+        {
+            ChallengeName = ChallengeNameType.SOFTWARE_TOKEN_MFA,
+            ClientId = clientId,
+            ChallengeResponses = challengeResponses,
+            Session = session
+        };
+
+        var response = await _cognitoService.RespondToAuthChallengeAsync(respondToAuthChallengeRequest);
+        Console.WriteLine($"Response to Authentication {response.AuthenticationResult}");
+        return response.AuthenticationResult;
+    }
+
+    // snippet-end:[Cognito.dotnetv3.RespondToAuthChallenge]
+
+    // snippet-start:[Cognito.dotnetv3.VerifySoftwareToken]
+    /// <summary>
+    /// Verify the TOTP and register for MFA.
+    /// </summary>
+    /// <param name="session"></param>
+    /// <param name="code"></param>
```

Comment:
Are there missing values here, or intentionally blank? There are a few like this throughout the file.

Reply:
Not intentional. Filled them in and will check the rest of the files.